examples/bigquery-analyze-realtime-reddit-data/terraform/main.tf

```diff
@@ -0,0 +1,284 @@
+# service account for reddit VM
+resource "google_service_account" "reddit_service_account" {
+  account_id   = "${var.service_account_name}"
+  display_name = "Reddit VM Service Account"
+}
+
+resource "google_project_iam_binding" "reddit_vm_compute" {
+  project = "${var.project_id}"
+  role    = "roles/compute.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+resource "google_project_iam_binding" "reddit_vm_dataflow" {
+  project = "${var.project_id}"
+  role    = "roles/dataflow.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+resource "google_project_iam_binding" "reddit_vm_pubsub" {
+  project = "${var.project_id}"
+  role    = "roles/pubsub.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+
+resource "google_compute_instance" "reddit_vm" {
+  name         = "reddit-data-collector-1"
+  machine_type = "e2-micro"
+  zone         = "us-central1-a"
```

Comment:
would be cool to have that in the vars, feel free to use us-central1-a as a default